app/src/main/java/com/tyranor/next/ui/pages/GameScreen.kt

```
@@ -1037,8 +1258,65 @@ private fun decodeCoverThumbnail(context: android.content.Context, uriText: Stri
     context.contentResolver.openInputStream(uri)?.use { BitmapFactory.decodeStream(it, null, options) }
 }.getOrNull()
 
+private fun decodeRemoteCoverThumbnail(urlText: String, source: String): Bitmap? = runCatching {
```

detekt (1.23.8)

[warning] The function decodeRemoteCoverThumbnail appears to be too complex based on Cognitive Complexity (complexity: 31). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)